torchtune/modules/tokenizers/_sentencepiece.py

```diff
@@ -60,19 +59,21 @@ def encode(
 
         Args:
             text (str): The input text to be encoded, unbatched.
-            add_bos (bool): Whether to prepend BOS special token (Beginning of Sentence) to the input, defaults to True.
-            add_eos (bool): Whether to append EOS special token (End of Sentence) to the input, defaults to True.
+            add_bos (bool): Whether to prepend BOS to the input, defaults to True.
+            add_eos (bool): Whether to append EOS to the input, defaults to True.
             trim_leading_whitespace (bool): Whether to trim leading whitespace from
                 underlying sentencepiece tokenization. Sentencepiece normally prepends
                 whitespace to any tokenized text, which can cause differences where
                 encode(s1) + encode(s2) != encode(s1 + s2) due to leading whitespace
-                added to s2. Default: False
+                added to s2. This will only trim leading whitespace if the underlying
+                ``SentencePieceProcessor`` encodes whitespace. Default: False
             prefix (Optional[str]): Optional string to encode for trimming leading
                 whitespaces. Used only if trim_leading_whitespace=True. Default: None
+
         Returns:
             List[int]: The encoded token IDs.
         """
-        if trim_leading_whitespace:
+        if trim_leading_whitespace and self.encodes_whitespace:
```

Comment:
The rationale here is also less clear now that we've split encode and tokenize_messages. Please make sure to document why we're doing this